### PR TITLE
[WFLY-12520] Replace managed executor config's thread-factory attribute with thread-priority

### PR DESCRIPTION
**Pull request by emmartins**
Issues: https://issues.jboss.org/browse/WFLY-12520, https://issues.jboss.org/browse/WFLY-12467

### CODE REVIEW

**Comment by emmartins** [comment]
fyi Includes "model bump" commit common to #12567

File: ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceResourceDefinition.java [hunk 1]
```diff
@@ -91,6 +93,15 @@
                     .setValidator(new StringLengthValidator(0, true))
                     .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
                     .setCapabilityReference(ManagedThreadFactoryResourceDefinition.CAPABILITY.getName(), CAPABILITY)
+                    .setDeprecated(EESubsystemModel.Version.v5_0_0)
+                    .build();
+
+    public static final SimpleAttributeDefinition THREAD_PRIORITY_AD =
+            new SimpleAttributeDefinitionBuilder(THREAD_PRIORITY, ModelType.INT, true)
```
Comment (bstansberry) [inline ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceResourceDefinition.java]: This and THREAD_FACTORY_AD should both use setAlternatives to declare that each is an alternative to the other.

File: ee/src/main/java/org/jboss/as/ee/subsystem/ManagedScheduledExecutorServiceResourceDefinition.java [hunk 2]
```diff
@@ -84,6 +86,15 @@
                     .setValidator(new StringLengthValidator(0, true))
                     .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
                     .setCapabilityReference(ManagedThreadFactoryResourceDefinition.CAPABILITY.getName(), CAPABILITY)
+                    .setDeprecated(EESubsystemModel.Version.v5_0_0)
+                    .build();
+
+    public static final SimpleAttributeDefinition THREAD_PRIORITY_AD =
+            new SimpleAttributeDefinitionBuilder(THREAD_PRIORITY, ModelType.INT, true)
```
Comment (bstansberry) [inline ee/src/main/java/org/jboss/as/ee/subsystem/ManagedScheduledExecutorServiceResourceDefinition.java]: This and THREAD_FACTORY_AD should both use setAlternatives to declare that each is an alternative to the other.

**Review by bstansberry — CHANGES_REQUESTED** [review]
@emmartins Small issues in my comments aside, looks good. Please @ ping me when corrected and we'll get this pushed through. Thanks for your patience.

**Comment by emmartins** [comment]
thanks @bstansberry, requested changes applied

**Review by bstansberry — DISMISSED** [review]

**Comment by bstansberry** [comment]
@emmartins The tests failures are related.

**Comment by emmartins** [comment]
@bstansberry fixed

**Review by bstansberry — APPROVED** [review]

**Pull request merged**